x-pack/plugins/apm/public/hooks/use_time_range.ts

```diff
@@ -5,19 +5,42 @@
  * 2.0.
  */
 
-import { useUrlParams } from '../context/url_params_context/use_url_params';
+import { isEqual } from 'lodash';
+import { useCallback, useRef, useState } from 'react';
+import { getDateRange } from '../context/url_params_context/helpers';
 
-export function useTimeRange() {
-  const {
-    urlParams: { start, end },
-  } = useUrlParams();
+export function useTimeRange({
+  rangeFrom,
+  rangeTo,
+}: {
+  rangeFrom: string;
+  rangeTo: string;
+}) {
+  const rangeRef = useRef({ rangeFrom, rangeTo });
 
-  if (!start || !end) {
-    throw new Error('Time range not set');
+  const [timeRangeId, setTimeRangeId] = useState(0);
+
+  const stateRef = useRef(getDateRange({ state: {}, rangeFrom, rangeTo }));
+
+  const updateParsedTime = useCallback(() => {
+    stateRef.current = getDateRange({ state: {}, rangeFrom, rangeTo });
```

Comment:
Should `state` always be empty? Looks like will then always re-calculate the date.
Also: is there anyway we can simplify this logic?

Reply:
I'm reusing the function from `useUrlParams`. It intentionally recalculates in this case. Once we deprecate `useUrlParams` we can at least clean `getDateRange` up. I'm not sure if we can simplify this further, as we have to maintain state, but open to suggestions.